### PR TITLE
LESS: icon mixin consolidation

### PR DESCRIPTION
**Pull request by ianmcburnie**
This PR reduces the number of icon related mixins (both utility and specific) down to the following:

```less
// base styles for BEM span.icon block
.background-icon-base-mixin() {
    background-repeat: no-repeat;
    display: inline-block;
    vertical-align: middle;
}

// base styles for BEM svg.icon block
.foreground-icon-base-mixin() {
    display: inline-block;
    fill: currentColor;
    pointer-events: none;
    stroke: currentColor;
    stroke-width: 0;
    vertical-align: middle;
}

.background-icon-image-mixin(@name) {
    background-image: url('data:image/svg+xml;base64,@{icon-@{name}-base64}');
    height: ~"@{icon-@{name}-height}"; // string resolves to a LESS variable
    width: ~"@{icon-@{name}-width}"; // string resolves to a LESS variable
}

// LESS extension of .background-icon-image-mixin
.background-icon-image-mixin(@name, @width, @height) {
    background-image: url('data:image/svg+xml;base64,@{icon-@{name}-base64}');
    height: ~"@{height}"; // string resolves to a LESS variable
    width: ~"@{width}"; // string resolves to a LESS variable
}

// LESS extension of .background-icon-image-mixin
.background-icon-image-mixin(@name; @width; @height) when (@width >= 15) and (@height >= 15) {
    background-size: contain;
}

// LESS extension of .background-icon-image-mixin
// https://github.com/eBay/skin/issues/334
.background-icon-image-mixin(@name; @width; @height) when (@width < 15), (@height < 15) {
    background-size: @width @height;
}

.foreground-icon-image-mixin(@name) {
    height: ~"@{icon-@{name}-height}"; // string resolves to a LESS variable
    width: ~"@{icon-@{name}-width}"; // string resolves to a LESS variable
}

// shorthand for setting base64
.background-image-base64(@base64) {
    background-image: url('data:image/svg+xml;base64,@{base64}');
}
```

Any background icon can now be constructed like so:

```less
span {
  .background-icon-base-mixin;
  .background-icon-image-mixin('play');
}
```

For foreground (svg):

```less
svg {
  .foreground-icon-base-mixin;
  .foreground-icon-image-mixin('play');
}
```

I would have preferred not to have the word "mixin" in the names, but I've added it to be very clear and specific, but mainly to avoid potential clashes with the old names in app code. In version 9 we could remove it.

The big change is that I have removed all the specific icon mixins, e.g. `.icon-play, .icon-confirmation, etc`. Reason being that these can easily be constructed using the method above and means less maintenance for us. If app teams cry out for these we can consider adding them back.

For consistency, I also updated everywhere to use the pre-existing `.background-image-base64` mixin.

Dist files have not been impacted (bar one change to notice.css - which seems valid) - which is what I would expect (and hoped for!).

### CODE REVIEW

File: dist/notice/ds4/notice.css [hunk 1]
```diff
@@ -319,19 +319,16 @@ button.page-notice__close span {
   background-image: url('data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAyNCAyNCI+PHBhdGggZmlsbD0iI2RkMWUzMSIgZD0iTTEyLjAwNyAxNC4yNjFhLjg0OS44NDkgMCAwIDEtLjg0OS0uODQ5VjYuODU5YS44NDkuODQ5IDAgMSAxIDEuNjg0LjAwNXY2LjU0OWEuODQ5Ljg0OSAwIDAgMS0uODM0Ljg0OWwtLjAwMS0uMDAxem0xLjA1NyAyLjg5NGExLjAxNiAxLjAxNiAwIDEgMS0yLjAzMiAwIDEuMDE2IDEuMDE2IDAgMCAxIDIuMDMyIDB6bS0xLjA1NyA2Ljg1OUM1LjM3NiAyNC4wMTQgMCAxOC42MzggMCAxMi4wMDcgMCA1LjM3NiA1LjM3NiAwIDEyLjAwNyAwYzYuNjMxIDAgMTIuMDA3IDUuMzc2IDEyLjAwNyAxMi4wMDd2LjAxNWMtLjAwOCA2LjYyNS01LjM4MSAxMS45OTItMTIuMDA3IDExLjk5MnptMC0yMi4yNjFjLTUuNjcxIDAtMTAuMjY4IDQuNTk3LTEwLjI2OCAxMC4yNjhzNC41OTcgMTAuMjY4IDEwLjI2OCAxMC4yNjggMTAuMjY4LTQuNTk3IDEwLjI2OC0xMC4yNjhTMTcuNjc4IDEuNzUzIDEyLjAwNyAxLjc1M3oiLz48L3N2Zz4=');
   height: 24px;
   width: 24px;
-  background-size: contain;
```
Comment (agliga) [inline dist/notice/ds4/notice.css]: Should this be missing in notice?

Comment (ianmcburnie) [reply dist/notice/ds4/notice.css]: I believe that change to notice is correct, since the height and width are over 24px.

**Comment by ianmcburnie** [comment]
@sendlo Good spot! We are inconsistent across the board on this. I wonder if it's worth just leaving as is then creating a separate ticket for this so we can do them all in one go?

File: src/less/less/mixins-common.less [hunk 2]
```diff
@@ -1,41 +1,5 @@
 @import "variables-common.less";
```
Comment (sendlo) [inline src/less/less/mixins-common.less]: Looks like we can remove this import now.

Comment (ianmcburnie) [reply src/less/less/mixins-common.less]: Yeah, to be honest, you'll probably see a bunch of redundant imports like this. Another of those things where we'll need to sweep through all files at some point and get consistency of our headers/imports.

**Comment by ianmcburnie** [comment]
@sendlo Yep. I think we can take care of those later. Feel free to create a ticket for the LESS parenthesis issue.

**Pull request merged**